### PR TITLE
Add revealTerminal and focusInstance in terminalService

### LINKED ISSUE #221464: Running a task does not show the build terminal if an editor terminal is open
Type: <b>Bug</b>

Preparation: create the following `.vscode/tasks.json` file:
```json
{
  "version": "2.0.0",
  "tasks": [
    {
      "label": "Test",
      "type": "shell",
      "command": "sleep 10"
    }
  ]
}
```

1. In the terminals panel, create a new shell terminal
2. Drag it to the main editor area to make it a full-size terminal
3. Also open an editor tab (any file will do, including Command-N)
4. Make sure the panel is closed (press Command-J to close it if it is open)
5. Run the above task (e.g. with Command-P, then type `task Test`)

Expected: the panel opens and shows the running task
Actual: the editor terminal is brought to the front, and the panel stays closed

VS Code version: Code - Insiders 1.92.0-insider (c37f4a49e28831cec1aec1af0f626edaada1f7ce, 2024-07-11T05:03:58.450Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 Max (10 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|4, 5, 6|
|Memory (System)|32.00GB (3.14GB free)|
|Process Argv|--disable-extensions . --crash-reporter-id 0af73232-1e42-4c3f-95c4-5b10f8deb228|
|Screen Reader|no|
|VM|0%|
</details>Extensions disabled

### PR DESCRIPTION
**Pull request by tisilent**
fix #221464

because:
https://github.com/microsoft/vscode/blob/cf08702ebbc5baa015ef522e2b2af6f057ff31c6/src/vs/workbench/contrib/terminal/browser/terminalService.ts#L561-L571
https://github.com/microsoft/vscode/blob/cf08702ebbc5baa015ef522e2b2af6f057ff31c6/src/vs/workbench/contrib/terminal/browser/terminalService.ts#L111-L122

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts [hunk 1]
```diff
@@ -558,18 +558,26 @@ export class TerminalService extends Disposable implements ITerminalService {
 		return instance;
 	}
 
-	async revealActiveTerminal(preserveFocus?: boolean): Promise<void> {
-		const instance = this.activeInstance;
-		if (!instance) {
+	async revealTerminal(source: ITerminalInstance, preserveFocus?: boolean): Promise<void> {
+		if (!source) {
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalService.ts:562]: is it possible that `source` is undefined?

I think we can remove this check?

**Review by meganrogge — CHANGES_REQUESTED** [review]
Thanks for investigating this, LGTM, just one comment

File: src/vs/workbench/contrib/terminal/browser/terminalService.ts (hunk 1 above)
Comment (tisilent) [reply src/vs/workbench/contrib/terminal/browser/terminalService.ts:562]: You're right.

**Comment by tisilent** [comment]
https://github.com/microsoft/vscode/blob/06f8ef165cc37104bc78b7f36209d95333474348/src/vs/workbench/contrib/tasks/browser/terminalTaskSystem.ts#L1050-L1052
The same problem.

**Review by meganrogge — APPROVED** [review]
Thanks!

**Review by Tyriar — APPROVED** [review]

**Pull request merged**